### PR TITLE
[azure-core] Small fixes for aiohttp

### PR DESCRIPTION
**Pull request by annatisch**
There are 3 fixes in this PR:
- Aiohttp does not define an error for request read timeouts - instead it relies on the async future to timeout. The patch I have here catches the asyncio.TimeoutError and raises the ServiceResponseError - however I'm not entirely sure this correct - given that there may be other reasons the future times-out. Either way, whichever is raised here, the retry policy must be updated to work with it.
- Aiohttp ClientResponse object has a "status" attribute rather than the "status_code" used by Requests.
- Aiohttp uses a single string for the content-type headers rather than the list used by Requests.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/core/azure-core/azure/core/pipeline/policies/universal.py [hunk 1]
```diff
@@ -350,7 +350,10 @@ def deserialize_from_http_generics(cls, response):
         # Try to use content-type from headers if available
         content_type = None
         if response.content_type: # type: ignore
-            content_type = response.content_type[0].strip().lower() # type: ignore
+            try:
+                content_type = response.content_type.strip().lower() # type: ignore
+            except AttributeError:
+                content_type = response.content_type[0].strip().lower() # type: ignore
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/pipeline/policies/universal.py]: My only suggestion really is that the try/except blocks here and above could go in their own functions e.g. `get_status_code`, `get_content_type` to keep this kind of conditioning based off implementation more hidden away

Comment (annatisch) [reply sdk/core/azure-core/azure/core/pipeline/policies/universal.py]: Fixed!

**Comment by annatisch** [comment]
Added another fix to this PR - found by @iscai-msft 
There was an issue with a couple of the async transports inheriting the synchronous sleep function which then failed when called as a coroutine (e.g. in the async retry policy).

Sorry to add another feature @bryevdv ....

**Comment by annatisch** [comment]
@chlowell - I remember at one point we put handling into azure-core because aiohttp (iirc) was returning the `content-type` header as a list rather than a string. This no longer seems to be the case... can you recall the particulars? Or am I getting completely confused with something else?

File: sdk/core/azure-core/azure/core/pipeline/transport/requests_trio.py [hunk 2]
```diff
@@ -143,7 +140,7 @@ class TrioRequestsTransport(RequestsTransport, AsyncHttpTransport):  # type: ign
         return super(TrioRequestsTransport, self).__exit__()
 
     async def sleep(self, duration):
-        await trio.sleep(duration)
+        return super(TrioRequestsTransport, self).sleep(duration)
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/transport/requests_trio.py]: seems a surprise?

**Comment by lmazuel** [comment]
So CI, is doing something really weird:
- All Python 3.5 tests are passing for "identity" CI
- Python 3.5 Windows passes for all tests, but linux fails for "identity" tests
- Everything else passes in all platform and Python

I believe it's an issue in the CI :(

**Pull request merged**